### PR TITLE
fix keep order in table scan is lost when remote read of PartitionTableScan

### LINKED ISSUE #7519: running ch partition workload and meets error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. run ch list partition workload
2. check consistency between tiflash and tikv

### 3. What did you see instead (Required)
[2023/05/21 23:25:24.121 +08:00] [ERROR] [ch.go:256] ["check warehouse 2 at condition [3.3.2.5](http://3.3.2.5/) failed exec SELECT count(*)  FROM orders LEFT
 JOIN new_order ON (no_w_id=o_w_id AND o_d_id=no_d_id AND o_id=no_o_id) where o_w_id = ? and ((o_carrier_id IS NULL and no_o_id IS  NULL) O
R (o_carrier_id IS NOT NULL and no_o_id IS NOT NULL  ))  failed Error 1105: other error for mpp stream: From MPP<query:<query_ts:1684682723
943347087, local_query_id:77237, server_id:2001687, start_ts:441629467975614567>,task_id:4>: Code: 0, e.displayText() = DB::Exception: Code
: 0, e.displayText() = DB::Exception: code: 49\nmsg: \"Check !filter || !filter->before_where || read_mode == ReadMode::Bitmap failed: Push
 down filters needs bitmap: (while creating InputStreams from storage `db_31987`.`t_34126`, table_id: 34126)\"\n, e.what() = DB::Exception,
"] [stack="[github.com/pingcap/endless/testcase/tiflash/pkg/workload.(*ChWorkload).checkConsistence\n\t/home/jenkins/agent/workspace/endless](http://github.com/pingcap/endless/testcase/tiflash/pkg/workload.(*ChWorkload).checkConsistence/n/t/home/jenkins/agent/workspace/endless)
-master-build/testcase/tiflash/pkg/workload/ch.go:256\[ngithub.com/pingcap/endless/testcase/tiflash/pkg/workload.(*ChWorkload).RunTP.func3\n](http://ngithub.com/pingcap/endless/testcase/tiflash/pkg/workload.(*ChWorkload).RunTP.func3/n)
\t/home/jenkins/agent/workspace/endless-master-build/testcase/tiflash/pkg/workload/ch.go:208\[ngolang.org/x/sync/errgroup.(*Group).Go.func1](http://ngolang.org/x/sync/errgroup.(*Group).Go.func1)\
n\t/go/pkg/mod/[golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75](http://golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75)"]
[2023/05/21 23:27:51.519 +08:00] [ERROR] [ch.go:256] ["check warehouse 7 at condition 3.3.2.10 failed exec SELECT count(*) \n\tFROM (  SELE
CT  c.c_id, c.c_d_id, c.c_w_id, c.c_balance c1, \n\t\t\t\t   (SELECT sum(ol_amount) FROM orders STRAIGHT_JOIN order_line \n\t\t\t\t\t WHERE
 OL_W_ID=O_W_ID \n\t\t\t\t\t   AND OL_D_ID = O_D_ID \n\t\t\t\t\t   AND OL_O_ID = O_ID \n\t\t\t\t\t   AND OL_DELIVERY_D IS NOT NULL \n\t\t\t
\t\t   AND O_W_ID=? \n\t\t\t\t\t   AND O_D_ID=c.C_D_ID \n\t\t\t\t\t   AND O_C_ID=c.C_ID) sm, (SELECT  sum(h_amount)  from  history \n\t\t\t
\t\t\t\t\t\t\t\t\tWHERE H_C_W_ID=? \n\t\t\t\t\t\t\t\t\t\t\t\t  AND H_C_D_ID=c.C_D_ID \n\t\t\t\t\t\t\t\t\t\t\t\t  AND H_C_ID=c.C_ID) smh \n\
t\t\t FROM customer c \n\t\t\tWHERE  c.c_w_id = ? ) t\n   WHERE c1<>sm-smh failed Error 1105: other error for mpp stream: From MPP<query:<q
uery_ts:1684682861658032567, local_query_id:77264, server_id:2001687, start_ts:441629504072843366>,task_id:10>: Code: 0, e.displayText() =
DB::Exception: Code: 0, e.displayText() = DB::Exception: code: 49\nmsg: \"Check !filter || !filter->before_where || read_mode == ReadMode::
Bitmap failed: Push down filters needs bitmap: (while creating InputStreams from storage `db_31987`.`t_34636`, table_id: 34636)\"\n, e.what
() = DB::Exception,"] [stack="github.com/pingcap/endless/testcase/tiflash/pkg/workload.(*ChWorkload).checkConsistence\n\t/home/jenkins/agen
t/workspace/endless-master-build/testcase/tiflash/pkg/workload/ch.go:256\ngithub.com/pingcap/endless/testcase/tiflash/pkg/workload.(*ChWork
load).RunTP.func3\n\t/home/jenkins/agent/workspace/endless-master-build/testcase/tiflash/pkg/workload/ch.go:208\ngolang.org/x/sync/errgroup
.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75"]
### 4. What is your TiFlash version? (Required)
sh-5.1# ./tiflash --version
TiFlash
Release Version: v7.1.0
Edition:         Community
Git Commit Hash: 70200bcd6d32f29faa9073a3e0a19c849abef241
Git Branch:      heads/refs/tags/v7.1.0
UTC Build Time:  2023-05-12 23:23:30
Enable Features: jemalloc sm4(GmSSL) avx2 avx512 unwind thinlto
Profile:         RELWITHDEBINFO

Raft Proxy
Git Commit Hash:   5cdf47726c58e04683c2da3e2892c276fccb3e8a
Git Commit Branch: HEAD
UTC Build Time:    2023-05-12 23:32:47
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Storage Engine:    tiflash
Prometheus Prefix: tiflash_proxy_
Profile:           release
Enable Features:   Unknown (env var does not exist when building)

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: close #7519

Problem Summary:

image

2307d72c-56af-4e8d-bf4b-41f4ce19b129

keep order sometimes lost.

### What is changed and how it works?

set keep order to false when remote read of PartitionTableScan

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by windtalker — APPROVED** [review]
LGTM

File: dbms/src/Flash/Coprocessor/TiDBTableScan.cpp [hunk 1]
```diff
@@ -79,6 +81,7 @@ void TiDBTableScan::constructTableScanForRemoteRead(tipb::TableScan * tipb_table
         for (auto id : partition_table_scan.primary_prefix_column_ids())
             tipb_table_scan->add_primary_prefix_column_ids(id);
         tipb_table_scan->set_is_fast_scan(partition_table_scan.is_fast_scan());
+        tipb_table_scan->set_keep_order(false);
```
Comment (JaySon-Huang) [inline dbms/src/Flash/Coprocessor/TiDBTableScan.cpp:84]: We could also pick this change back to release-6.5 branch

**Review by SeaRise — APPROVED** [review]

**Pull request merged**